router.go

```diff
@@ -0,0 +1,177 @@
+
+	// roles are for listing the roles
+	if !r.config.Debug {
+		internal.GET("/roles", r.views.RolesFunc, r.views.RequirePermission(permissions.ManageMembersGroup))
+	} else {
+		internal.GET("/roles", r.views.RolesFunc)
+	}
+
+	role := internal.Group("/role")
+	// role is any function to do with a specific role or new role
+	if !r.config.Debug {
+		role.Use(r.views.RequirePermission(permissions.ManageMembersGroup))
+	}
+	role.Match(validMethods, "/add", r.views.RoleAddFunc)
+	roleID := role.Group("/:roleid")
+	// roleID is any function to do with a specific role
+	roleID.Match(validMethods, "/edit", r.views.RoleEditFunc)
+	roleID.Match(validMethods, "/delete", r.views.RoleDeleteFunc)
+	roleID.Match(validMethods, "", r.views.RoleFunc)
+
+	// this section of users is a bit weird, users is valid for anyone who can list users and user/add can be used by add users permission
+	if !r.config.Debug {
+		internal.Match(validMethods, "/users", r.views.UsersFunc, r.views.RequirePermission(permissions.ManageMembersMembersList))
+		internal.Match(validMethods, "/user/add", r.views.UserAddFunc, r.views.RequirePermission(permissions.ManageMembersMembersAdd))
+	} else {
+		internal.Match(validMethods, "/users", r.views.UsersFunc)
+		internal.Match(validMethods, "/user/add", r.views.UserAddFunc)
+	}
+
+	user := internal.Group("/user/:userid")
+	// user is any function to do with a specific user
+	if !r.config.Debug {
+		user.Use(r.views.RequirePermission(permissions.ManageMembersMembersAdmin))
+	}
+	user.Match(validMethods, "/edit", r.views.UserEditFunc)
+	user.Match(validMethods, "/delete", r.views.UserDeleteFunc)
+	user.Match(validMethods, "/reset", r.views.ResetUserPasswordFunc)
+	user.Match(validMethods, "", r.views.UserFunc)
+
+	api := r.router.Group("/api")
+	// api is all the methods that are used by the api interactions
+	api.GET("/set_token", r.views.SetTokenHandler)
+	api.GET("/test", r.views.TestAPITokenFunc)
+	api.GET("/health", func(c echo.Context) error {
+		marshal, err := json.Marshal(struct {
+			Status int `json:"status"`
+		}{
+			Status: http.StatusOK,
+		})
+		if err != nil {
+			fmt.Println(err)
+			return &echo.HTTPError{
+				Code:     http.StatusBadRequest,
+				Message:  err.Error(),
+				Internal: err,
+			}
+		}
+
+		c.Response().Header().Set("Content-Type", "application/json")
+		return c.JSON(http.StatusOK, marshal)
+	})
+
+	base := r.router.Group("/")
+	// base is the functions that don't require being logged in
+	base.GET("", r.views.IndexFunc)
+	base.Match(validMethods, "login", r.views.LoginFunc)
+	base.Match(validMethods, "logout", r.views.LogoutFunc)
+	base.Match(validMethods, "signup", r.views.SignUpFunc)
+	base.Match(validMethods, "forgot", r.views.ForgotFunc)
+	base.Match(validMethods, "reset/:url", r.views.ResetURLFunc)
+}
+
+func getFileSystem() http.FileSystem {
+	fsys, err := fs.Sub(embeddedFiles, "public/static")
+	if err != nil {
+		panic(err)
+	}
+
+	return http.FS(fsys)
+}
```

Comment:
Echo has a [`MustSubFS`](https://pkg.go.dev/github.com/labstack/echo/v4#MustSubFS) helper that does exactly this